crates/biome_formatter/src/trivia.rs

```diff
@@ -37,14 +66,23 @@ where
             FormatLeadingComments::Comments(comments) => comments,
         };
 
-        for comment in leading_comments {
+        for (index, comment) in leading_comments.iter().enumerate() {
```

Comment:
No idea what the best way to do this is. It _feels_ awkward to have to directly get the iterated and then `enumerate` it, but it's probably the most efficient way to get both the current and the next item? This is my lack of Rust experience coming through.

Reply:
If you need both the current and next item, you can use [`peekable`](https://doc.rust-lang.org/std/iter/struct.Peekable.html).

```rs
let mut leading_comments_iter = leading_comments.iter().peekable();
for comment in leading_comments_iter {
  let next_comment = leading_comments_iter.peek();
}
```